internal/policy/policy.go

```diff
@@ -8,3 +10,23 @@ const (
 	PolicyScratch   Policy = "scratch"
 	PolicyRoot      Policy = "root"
 )
+
+// ContextWithPolicy adds Policy p to the context ctx.
+func ContextWithPolicy(ctx context.Context, p Policy) context.Context {
+	return context.WithValue(ctx, policyContextKey, p)
+}
+
+// PolicyFromContext returns the policy from the context, or empty string.
+func PolicyFromContext(ctx context.Context) Policy {
```

Comment:
I think I'd prefer something like...

```
policy.NewContext(ctx, p)
```

and just 

```
policy.FromContext(ctx)
```


Otherwise, you're saying

```
policy.PolicyFromContext
```

It's redundant.

Reply:
I also realize that artifacts should be like this too...

Reply:
Makes, sense. I've updated this.